tools/mapping/gen_mapping_files/gen_ESMF_mapping_file/create_ESMF_map.sh

```diff
@@ -352,6 +357,7 @@ if [ ! -z $ESMFBIN_PATH ]; then
 else
   ESMF_REGRID="ESMF_RegridWeightGen"
 fi
+ESMF_REGRID="$ESMF_REGRID --ignore_degenerate "
```

Comment:
I think this argument should be hard-coded in line 403

```diff
-cmd="$MPIEXEC $ESMF_REGRID --ignore_unmapped -m $mapping -w $fmap -s $fsrc -d $fdst $pass_thru"
+cmd="$MPIEXEC $ESMF_REGRID --ignore_degenerate --ignore_unmapped -m $mapping -w $fmap -s $fsrc -d $fdst $pass_thru"
```

We definitely don't need it before the `command -v` call in the next block (which just makes sure the executable exists)

Reply:
Yeah, put it with the other hardcoded settings. That makes sense.

I think the flag is something that we want to always use. The ESMF people recommended it for particular work I was doing, but I don't see a reason not to have it on all the time.

Reply:
This is the comment about --ignore_degenerate in ESMF:

>  --ignore_degenerate - ignore degenerate cells in the input grids. If not specif
>  ied,
>                            the default is to stop with an error.

Reply:
@mnlevy1981 I pushed some changes, let me know if that's better now. I didn't test, but it's straightforward enough that it shouldn't be a problem. And I will at least run a test for it.

Reply:
I did run a simple test. I give details on it in regular discussion.